File: 2255-minimum-swaps-to-group-all-1s-together-ii/minimum-swaps-to-group-all-1s-together-ii.c
int minSwaps(int* nums, int numsSize) {
    int o = 0;

    for (int i = 0; i < numsSize; i++) {
        o += nums[i];
    }

    if (o == 0) {
        return 0;
    }

    int currentSum = 0;
    for (int i = 0; i < o; i++) {
        currentSum += nums[i];
    }

    int minSwaps = o - currentSum;

    for (int i = 1; i < numsSize; i++) {
        currentSum = currentSum - nums[i - 1] + nums[(i + o - 1) % numsSize];
        if (o - currentSum < minSwaps) {
            minSwaps = o - currentSum;
        }
    }

    return minSwaps;
}

File: 2255-minimum-swaps-to-group-all-1s-together-ii/minimum-swaps-to-group-all-1s-together-ii.py
class Solution(object):
    def minSwaps(self, nums):
        n = len(nums)
        o = sum(nums)
        if o == 0:
            return 0
        
        # Initial window sum of the first 'o' elements
        current_sum = sum(nums[:o])
        min_swaps = o - current_sum

        for i in range(1, n):
            current_sum -= nums[i - 1]
            current_sum += nums[(i + o - 1) % n]
            min_swaps = min(min_swaps, o - current_sum)
        
        return min_swaps
